processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -59,9 +59,9 @@ type WatchClient struct {
 	Namespaces map[string]*Namespace
 }
 
-// Extract deployment name from the pod name. Pod name is created using
-// format: [deployment-name]-[Random-String-For-ReplicaSet]-[Random-String-For-Pod]
-var dRegex = regexp.MustCompile(`^(.*)-[0-9a-zA-Z]*-[0-9a-zA-Z]*$`)
+// Extract replicaset name from the pod name. Pod name is created using
+// format: [deployment-name]-[Random-String-For-ReplicaSet]
+var rRegex = regexp.MustCompile(`^(.*)-[0-9a-zA-Z]*$`)
```

Comment:
```suggestion
var rRegex = regexp.MustCompile(`^(.*)-[0-9a-zA-Z]+$`)
```

I don't believe all these symbols are really being used, but at least let's make sure we require the suffix for the match